### ISSUE TITLE
Using watcher causes changes to generate new instance of elements on DOM rather than replacing old ones

### ISSUE THREAD

**Issue by jfred1979** [fuse-box/fuse-box#1980]
fuse.ts looks like this:

`import { fusebox } from 'fuse-box';
const fuse = fusebox({
  entry: 'src/index.ts',
  target: 'browser',
  devServer: { enabled: true, open: true },
  webIndex: true,
  watcher: true,
  cache: true,
});
fuse.runDev();`

If I add an element to the DOM at runtime while the dev server and watcher are running, I just get another instance of the DOM element on screen. I would expect the DOM to refresh with whatever changes I made. For example, having in this in my index.ts:

`const el:HTMLElement = document.createElement('button');
document.body.appendChild(el);`

causes a new button to appear on screen every time I save the file. 

Is this a cacheing issue of some sort? Is my fuse config setup incorrectly?

Currently using latest Chrome on a Mac for testing.

**Comment by nchanged** [comment]
There is nothing wrong, the entry point is being triggered this is expected. YOu need to handle it this soft reload at your application level. For example
```
if(window.initialized) {
  clearAndStart()
} else {
  window.initialized = true;
  justStart()
}
```

**Comment by jfred1979** [comment]
Ah, okay. I suppose other bundlers that handle this out of the box probably have some kind of logic automatically added to the container HTML page? In my case the project is very simple and a full page refresh between saves is sufficient. So I guess I really just need live reloading, not hot reloading. Is there really no option for automatic live reloading right out of the box?

**Comment by nchanged** [comment]
It's not that a bundler "handles" them. It's framework specific.  As I said, the following code will re-init your app and no refresh needed.
```
if(window.initialized) {
 document.body.innerHTML = ""
} 
window.initialized = true;
const el:HTMLElement = document.createElement('button'); document.body.appendChild(el);

```

So once an entry point is triggered the second time you flush your state and start over. It's very handy.

Some more information for you:

https://fuse-box.org/docs/development/hmr

You could also take a look at the custom HMR plugin in this example:
https://github.com/fuse-box/sandbox/tree/master/hmr-plugin-example

https://github.com/fuse-box/sandbox/blob/master/hmr-plugin-example/src/hmr.ts

You can choose to reload the page too or apply some custom magic to it, good luck ;-)

**Comment by jfred1979** [comment]
Thanks, I edited my comment, you probably didn't see it in time. I did see this doc page:
https://fuse-box.org/docs/development/hmr
and the Typescript compiler seems to choke on 'hardReloadScripts' prop in the 'watch' or 'watcher' object (using v4.0.0 of fuse-box). I'm not clear if the documentation is for a previous version? 

Either way, for simple live reloading using the basic technique of checking the initialized prop and clearing the page in my dev builds should suffice for now.
